src/subdomains/core/buy-crypto/process/entities/buy-crypto-fees.entity.ts

```diff
@@ -39,32 +39,53 @@ export class BuyCryptoFee extends IEntity {
 
   //*** FACTORY METHODS ***//
 
-  static create(purchaseFeeAmount: number, payoutFeeAmount: number, transaction: BuyCrypto): BuyCryptoFee {
+  static create(
+    estimatePurchaseFeeAmount: number | null,
+    estimatePayoutFeeAmount: number | null,
+    transaction: BuyCrypto,
+  ): BuyCryptoFee {
     const entity = new BuyCryptoFee();
 
     entity.buyCrypto = transaction;
     entity.feeReferenceAsset = transaction.outputReferenceAsset;
 
-    entity.estimatePurchaseFeeAmount = purchaseFeeAmount;
-    entity.estimatePurchaseFeePercent = Util.round(purchaseFeeAmount / transaction.outputReferenceAmount, 8);
-    entity.estimatePayoutFeeAmount = payoutFeeAmount;
-    entity.estimatePayoutFeePercent = Util.round(payoutFeeAmount / transaction.outputReferenceAmount, 8);
+    entity.estimatePurchaseFeeAmount = estimatePurchaseFeeAmount;
+    entity.estimatePurchaseFeePercent =
+      estimatePurchaseFeeAmount != null
+        ? Util.round(estimatePurchaseFeeAmount / transaction.outputReferenceAmount, 8)
+        : null;
+
+    entity.estimatePayoutFeeAmount = estimatePayoutFeeAmount;
+    entity.estimatePayoutFeePercent =
+      estimatePayoutFeeAmount != null
+        ? Util.round(estimatePayoutFeeAmount / transaction.outputReferenceAmount, 8)
+        : null;
 
     return entity;
   }
 
   //*** PUBLIC API ***//
 
-  addActualPurchaseFee(purchaseFeeAmount: number, transaction: BuyCrypto): this {
-    this.actualPurchaseFeeAmount = purchaseFeeAmount;
-    this.actualPurchaseFeePercent = Util.round(purchaseFeeAmount / transaction.outputReferenceAmount, 8);
+  addActualPurchaseFee(purchaseFeeAmount: number | null, transaction: BuyCrypto): this {
+    if (purchaseFeeAmount == null) {
+      this.actualPurchaseFeeAmount = null;
+      this.actualPurchaseFeePercent = null;
+    } else {
+      this.actualPurchaseFeeAmount = purchaseFeeAmount;
+      this.actualPurchaseFeePercent = Util.round(purchaseFeeAmount / transaction.outputReferenceAmount, 8);
+    }
 
     return this;
   }
 
-  addActualPayoutFee(payoutFeeAmount: number, transaction: BuyCrypto): this {
-    this.actualPayoutFeeAmount = payoutFeeAmount;
-    this.actualPayoutFeePercent = Util.round(payoutFeeAmount / transaction.outputReferenceAmount, 8);
+  addActualPayoutFee(payoutFeeAmount: number | null, transaction: BuyCrypto): this {
+    if (payoutFeeAmount == null) {
+      this.actualPayoutFeeAmount = null;
+      this.actualPayoutFeePercent = null;
+    } else {
+      this.actualPayoutFeeAmount = payoutFeeAmount;
+      this.actualPayoutFeePercent = Util.round(payoutFeeAmount / transaction.outputReferenceAmount, 8);
+    }
```

Comment:
Can it still be null after payout is complete?

Reply:
same thing - yes, we can just fail to get the price for fee conversion

Reply:
![image](https://user-images.githubusercontent.com/85513542/208694834-cd777c1c-62e0-40de-99b1-ff99493f0721.png)
Ok, maybe it would make sense to not call this method, if fee is null? Do the check in BuyCrypto?

Reply:
no, I would rather have it explicitly set to NULL in addActualPayoutFee, than just omitting it in upper stack. Better describes intention that we want those props to be set to null, if fee is not available. imho